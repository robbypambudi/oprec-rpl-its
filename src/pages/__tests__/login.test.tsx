import '@testing-library/jest-dom';

import { act, fireEvent, render, screen } from '@testing-library/react';

import LoginPage from '@/pages/login/index.page';

jest.mock('next/router', () => require('next-router-mock'));

describe('Login Page', () => {
  it('renders correctly', () => {
    act(() => {
      render(<LoginPage />);
    });
    expect(screen.queryAllByText('Login')[0]).toBeInTheDocument();
    expect(screen.queryAllByText('Login')[0]).toBeVisible();

    expect(screen.queryAllByText('Email')[0]).toBeInTheDocument();
    expect(screen.queryAllByText('Email')[0]).toBeVisible();

    expect(screen.queryAllByText('Password')[0]).toBeInTheDocument();
    expect(screen.queryAllByText('Password')[0]).toBeVisible();

    expect(screen.queryAllByText('Login')[0]).toBeInTheDocument();
    expect(screen.queryAllByText('Login')[0]).toBeVisible();

    // Input Email
    act(() => {
      fireEvent.change(screen.queryAllByRole('textbox')[0], {
        target: { value: 'robby.pambudi10@gmail.com' },
      });
    });
    expect(screen.queryAllByRole('textbox')[0]).toHaveAttribute(
      'type',
      'email'
    );
    expect(screen.queryAllByRole('textbox')[0]).toHaveValue(
      'robby.pambudi10@gmail.com'
    );

    expect(screen.queryAllByText('Login')[0]).toMatchSnapshot();
    expect(screen.queryAllByRole('textbox')[0]).toMatchSnapshot();
  });
});
