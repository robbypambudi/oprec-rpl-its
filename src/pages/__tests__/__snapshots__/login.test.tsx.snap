// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page renders correctly 1`] = `
<h2
  class="text-3xl font-semibold md:text-3xl text-black text-center mb-5"
>
  Login
</h2>
`;

exports[`Login Page renders correctly 2`] = `
<input
  class="rounded-md px-4 py-2 border block w-full focus:border-primary-400"
  name="email"
  placeholder="Masukkan email"
  type="email"
/>
`;
